otsserver/stamper.py

```diff
@@ -50,33 +50,20 @@ def make_btc_block_merkle_tree(blk_txids):
     return digests[0]
 
 
-def make_timestamp_from_block(digest, block, blockheight, serde_txs, *, max_tx_size=500):
-    """Make a timestamp for a message in a block with cached serialized txs
-    see python-opentimestamps.bitcoin.make_timestamp_from_block
-    """
-    len_smallest_tx_found = max_tx_size + 1
-    commitment_tx = None
-    prefix = None
-    suffix = None
-    for (tx, serialized_tx) in serde_txs:
-
-        if len(serialized_tx) > len_smallest_tx_found:
-            continue
-
-        try:
-            i = serialized_tx.index(digest)
-        except ValueError:
-            continue
+def make_timestamp_from_block_tx(confirmed_tx, block, blockheight):
 
-        # Found it!
-        commitment_tx = tx
-        prefix = serialized_tx[0:i]
-        suffix = serialized_tx[i + len(digest):]
+    commitment_tx = confirmed_tx.tx
+    serialized_tx = commitment_tx.serialize(params={'include_witness': False})
+    digest = confirmed_tx.tip_timestamp.msg
 
-        len_smallest_tx_found = len(serialized_tx)
+    try:
+        i = serialized_tx.index(digest)
+    except ValueError:
+        logging.error("can't build a block_timestamp from my tx, this is not supposed to happen, exiting")
+        sys.exit(1)
```

Comment:
This should be done as an assertion, not a special bit of code. After all, sys.exit() might not be appropriate in all contexts.